modules/light-clients/08-wasm/types/update_test.go

```diff
@@ -449,208 +458,127 @@ func (suite *TypesTestSuite) TestUpdateStateGrandpa() {
-// 			}, true,
-// 		},
-// 		{
-// 			"invalid ClientMessage type", func() {
-// 				clientMessage = &ibctm.Misbehaviour{}
-// 			},
-// 			func() {},
-// 			false,
-// 		},
-// 	}
-// 	for _, tc := range testCases {
-// 		suite.Run(tc.name, func() {
-// 			suite.SetupWasmTendermint() // reset
-// 			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			callbackFn = func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, _ wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+				updateStateResp := types.UpdateStateResult{
+					Heights: []clienttypes.Height{},
+				}
 
-// 			err := path.EndpointA.CreateClient()
-// 			suite.Require().NoError(err)
+				resp, err := json.Marshal(updateStateResp)
+				if err != nil {
+					return nil, 0, err
+				}
 
-// 			// ensure counterparty state is committed
-// 			suite.coordinator.CommitBlock(suite.chainB)
-// 			clientMessage, height, err = path.EndpointA.Chain.ConstructUpdateWasmClientHeader(path.EndpointA.Counterparty.Chain, path.EndpointA.ClientID)
-// 			suite.Require().NoError(err)
+				return &wasmvmtypes.Response{
+					Data: resp,
+				}, types.DefaultGasUsed, nil
+			}
 
-// 			tc.malleate()
+			clientMsg = &types.ClientMessage{
+				Data: []byte{1},
+			}
 
-// 			clientState := path.EndpointA.GetClientState()
-// 			clientStore = suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), path.EndpointA.ClientID)
+			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
+			err := endpoint.CreateClient()
+			suite.Require().NoError(err)
 
-// 			if tc.expPass {
-// 				consensusHeights = clientState.UpdateState(suite.chainA.GetContext(), suite.chainA.App.AppCodec(), clientStore, clientMessage)
+			tc.malleate()
 
-// 				clientMessage, ok := clientMessage.(*types.ClientMessage)
-// 				suite.Require().True(ok)
-// 				var eHeader exported.ClientMessage
-// 				err := suite.chainA.Codec.UnmarshalInterface(clientMessage.Data, &eHeader)
-// 				tmHeader := eHeader.(*ibctm.Header)
-// 				suite.Require().NoError(err)
-// 				expTmConsensusState := &ibctm.ConsensusState{
-// 					Timestamp:          tmHeader.GetTime(),
-// 					Root:               commitmenttypes.NewMerkleRoot(tmHeader.Header.GetAppHash()),
-// 					NextValidatorsHash: tmHeader.Header.NextValidatorsHash,
-// 				}
-// 				wasmData, err := suite.chainA.Codec.MarshalInterface(expTmConsensusState)
-// 				suite.Require().NoError(err)
-// 				expWasmConsensusState := &types.ConsensusState{
-// 					Data: wasmData,
-// 				}
+			suite.mockVM.RegisterSudoCallback(types.UpdateStateMsg{}, callbackFn)
 
-// 				bz := clientStore.Get(host.ConsensusStateKey(height))
-// 				updatedConsensusState := clienttypes.MustUnmarshalConsensusState(suite.chainA.App.AppCodec(), bz)
+			clientState := endpoint.GetClientState()
 
-// 				suite.Require().Equal(expWasmConsensusState, updatedConsensusState)
+			var heights []exported.Height
```

Comment:
super nit or jw your thoughts, but if it's not super necessary to have a non-value assigned var declaration here, would it be better to just directly assign `heights := clientState.UpdateState`... in the if statement below: ```if tc.expPanic == nil {
				heights:= clientState.updateState()...} else { suite.RequirePanicsWithError...}```?